src/nvc.c

```diff
@@ -251,7 +252,42 @@ static int elaborate(int argc, char **argv)
          opt_set_str(OPT_DUMP_VCODE, optarg ?: "");
          break;
       case 'c':
-         opt_set_int(OPT_COVER, 1);
+         if (optarg) {
+            char prev = 0;
+            int n_chars = 0;
+            const char *full_cov_opt = optarg;
+            do {
+               if (*optarg == ',' || *optarg == '\0') {
+                  if (prev == 's')
+                     cover_mask |= COVER_MASK_STMT;
+                  else if (prev == 't')
+                     cover_mask |= COVER_MASK_TOGGLE;
+                  else if (prev == 'b')
+                     cover_mask |= COVER_MASK_BRANCH;
+                  else {
+                     diag_t *d = diag_new(DIAG_FATAL, NULL);
+                     diag_printf(d, "unknown coverage type '%c'", *optarg);
+                     diag_hint(d, NULL, "valid coverage types are: \n"
+                                        "  s (statement)\n"
+                                        "  t (toggle)\n"
+                                        "  b (branch)");
+                     diag_hint(d, NULL, "selected coverage types shall be "
+                                        "comma separated e.g $bold$--cover=s,t,b$$");
+                     diag_emit(d);
+                     fatal_exit(EXIT_FAILURE);
+                  }
+                  n_chars = 0;
+               }
+               n_chars++;
+               if (n_chars >= 3)
+                  fatal("Invalid coverage type: '%s'. Valid coverage shall be a "
```

Comment:
Just `"invalid coverage type '%'"` should be enough.

Reply:
Done